### ISSUE TITLE
Bug: no-useless-constructor rule doesn't work properly with NestJS DI

### ISSUE THREAD

**Issue by dkrylovdotcom** [typescript-eslint/typescript-eslint#5201]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Abug) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/docs/linting/troubleshooting) and my problem is not listed.

### Issue Description

Greetings, colleagues!

I faced with a problem of useless constructor with a rule: `@typescript-eslint/no-useless-constructor`

But in my case, it's false positive, coz [here](https://github.com/dkrylovdotcom/nestjs-di-eslint-problem/blob/master/src/validators/country-id.validator.ts#L12) I use NestJS DI capabilities to pass necessary parameters in the constructor.  
Yep, I understood, that is `eslint` thinkings that my `super(repository)` call is useless, but it's not true.

Any ideas? Is it a bug or I misunderstood something?

### Reproduction Repository Link

https://github.com/dkrylovdotcom/nestjs-di-eslint-problem

### Repro Steps

1) clone the repo
2) npm run lint

### Versions

| package                                | version |
| -------------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin`     | `5.28.0` |
| `@typescript-eslint/parser`            | `5.28.0` |
| `TypeScript`                           | `4.7.3` |
| `ESLint`                               | `8.17.0` |
| `node`                                 | `16.15.0` |

**Comment by bradzacher** [comment]
Nope - this rule just doesn't have support for treating directives on parameters as a marker for a useful constructor.
Haopy to accept a PR.

**Comment by armano2** [comment]
minimal reproduction case [Playground](https://typescript-eslint.io/play/#ts=4.7.2&sourceType=module&code=MYGwhgzhAEAqCmEAu14A8nwHYBMYHkAjAK3mBQG8BYAKGmmAHstkAnAV3MdYAoABAEJhW0AGaNGALmhsAllgDmASmjU69GewAO8XuMZKA3LXoBfWuZpA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6Jge1tmUSWWS0yHJsnxwy+DtHRRE0aNMjgwAXxCqgA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3QAacDUjoAHoizJ46IngCGqACL9YmNXl3Cwo9FIC+IM0A)

```ts
class Test extends Object {
  constructor(@Bar foo: string) {
    super(foo);
  }
}
```
